hub/api/dataset.py

```diff
@@ -19,6 +20,23 @@
 from hub.util.path import storage_provider_from_path
 
 
+def _get_cache_chain(path: str, storage: StorageProvider, memory_cache_size: int, local_cache_size: int, **kwargs):
+    if storage is not None and path:
+        warnings.warn(
+            "Dataset should not be constructed with both storage and path. Ignoring path and using storage."
+        )
+
+    if isinstance(storage, MemoryProvider):
+        return storage
```

Comment:
this function was moved from the __init__ function, the only new lines is the `isinstance` if statement, and this is to solve an error where 2 `MemoryProvider`s being passed to `generate_chain` means that when you call `storage.flush()` all data is deleted (presumably because of shared memory)? @AbhinavTuli was notified